core/payment-driver/zksync/src/zksync/wallet.rs

```diff
@@ -100,6 +100,43 @@ pub async fn make_transfer(details: &PaymentDetails) -> Result<String, GenericEr
     Ok(tx_hash)
 }
 
+pub async fn check_tx(tx_hash: &str) -> Option<bool> {
+    let provider = get_provider();
+    let tx_hash = format!("sync-tx:{}", tx_hash);
+    let tx_hash = TxHash::from_str(&tx_hash).unwrap();
+    let tx_info = provider.tx_info(tx_hash).await.unwrap();
+    log::debug!("tx_info: {:?}", tx_info);
+    // Check success
+    match tx_info.success {
+        None => None,
+        Some(result) => {
+            if result {
+                // Only wait for verified blocks on mainnet
+                if *NETWORK != Network::Mainnet || tx_info.is_verified() {
```

Comment:
Don't wait for verified blocks on mainnet either.

Reply:
👍